File: Find the Town Judge.cc
class Solution {
public:
    int findJudge(int N, vector<vector<int>>& trust) {
        set<int> possjudges;
        vector<int> votes(N+1, 0);
        
        
        for (int i = 1; i <= N; i ++) possjudges.insert(i);
        
        for (vector<int> v: trust) {
            int src = v[0];
            int tar = v[1];
            
            possjudges.erase(src);
            votes[tar] ++;
        }
        
        set<int>::iterator itr;
        
        for (itr = possjudges.begin(); itr != possjudges.end(); itr ++) {
            int tempj = *itr;
            
            if (votes[tempj] == N-1) return tempj;
        }
        
        return -1;
        
    }
};

File: Find the Town Judge.java
class Solution {
    public int findJudge(int N, int[][] trust) {
        int[] arr = new int[N+1];
            
        for(int i = 1 ; i<=N ; i++){
            arr[i] = 0;
        }
        for(int i = 0 ; i< trust.length; i++){
            int u = trust[i][0];
            int v = trust[i][1];
            arr[u]--;
            arr[v]++;
        }
        
        for(int i = 1 ;  i<=N ; i++){
            if(arr[i] == N-1){
                return i;
            }
        }
        return -1;
    }
}